devicetypes/qubino/qubino-flush-shutter.src/qubino-flush-shutter.groovy

```diff
@@ -0,0 +1,501 @@
+	def key = preference.key
+	def preferenceValue = getPreferenceValue(preference, cmd.scaledConfigurationValue)
+	if (settings."$key" == preferenceValue) {
+		state.currentPreferencesState."$key".value = settings."$key"
+		state.currentPreferencesState."$key".status = "synced"
+		handleConfigurationChange(cmd)
+	} else if (preference.type == "boolRange") {
+		if (state.currentPreferencesState."$key".status == "disablePending" && preferenceValue == preference.disableValue) {
+			state.currentPreferencesState."$key".status = "disabled"
+		} else {
+			runIn(5, "syncConfiguration", [overwrite: true])
+		}
+	} else {
+		state.currentPreferencesState."$key"?.status = "syncPending"
+		runIn(5, "syncConfiguration", [overwrite: true])
+	}
+	// Preferences template end
+	handleConfigurationChange(cmd)
+}
+
+private getPreferenceValue(preference, value = "default") {
+	def integerValue = value == "default" ? preference.defaultValue : value.intValue()
+	switch (preference.type) {
+		case "enum":
+			return String.valueOf(integerValue)
+		default:
+			return integerValue
+	}
+}
+
+private getCommandValue(preference) {
+	def parameterKey = preference.key
+	switch (preference.type) {
+		case "range":
+			return settings."$parameterKey"
+		default:
+			return Integer.parseInt(settings."$parameterKey")
+	}
+}
+
+private isPreferenceChanged(preference) {
+	if (settings."$preference.key" != null) {
+		def value = state.currentPreferencesState."$preference.key"
+		return state.currentPreferencesState."$preference.key".value != settings."$preference.key"
+	} else {
+		return false
+	}
+}
+
+def handleConfigurationChange(confgurationReport) {
+	switch (confgurationReport.parameterNumber) {
+		case 71: //Operating mode
+			switch (confgurationReport.scaledConfigurationValue) {
+				case 0: // Shutter
+					checkAndTriggerModeChange("windowShade")
+					break
+				case 1: // Venetian
+					checkAndTriggerModeChange("windowShadeVenetian")
+					break
+			}
+			log.info "Current device's mode is: ${state.currentMode}"
+			break
+		case 72:
+			state.timeOfVenetianMovement = confgurationReport.scaledConfigurationValue
+			break
+		default:
+			log.info "Parameter no. ${confgurationReport.parameterNumber} has no specific handler"
+			break
+	}
+}
+
+private checkAndTriggerModeChange(reportedMode) {
+	if (state.currentMode != reportedMode) {
+		state.currentMode = reportedMode
+		createVenetianBlindsChildDeviceIfNeeded()
+	}
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	} else {
+		log.warn "${device.displayName} - no-parsed event: ${description}"
+	}
+	log.debug "Parse returned: ${result}"
+	return result
+}
+
+def multilevelChildInstalled(childDni) {
+	state.timeOfVenetianMovement = 150
+	sendHubCommand(encap(zwave.switchMultilevelV3.switchMultilevelGet(), 2))
+}
+
+def close() {
+	setShadeLevel(0x64)
```

Comment:
why use `0x64` here if it's just getting converted to `0x63` later?

Reply:
I wanted be consistent with _setLevel_'s max value that can be set by user with mobile app (which is 100%, 0x64)